### PR TITLE
[gh-2473] add a tx sign subcommand.

### LINKED ISSUE #2473: Provide `rooch account sign_tx` or `rooch tx sign`? to sign the output tx from `rooch tx build`, and output tx hex or write to a file.

### PR DESCRIPTION
**Pull request by feliciss**
## Summary

`rooch tx sign` or `rooch transaction sign` command for signing the output from `rooch tx build`.

- Sign tx and output in command line

Sign tx with `--tx-hex`:

```zsh
rooch transaction sign --tx-hex 0ea3bb55c14aec8e074cd48c6a850526aa46965e176df53fadb433adeeeb56570000000000000000040000000000000000e1f5050000000001000000000000000000000000000000000000000000000000000000000000000305656d70747905656d7074790000 --output 
Sign transaction succeeded with the signed transaction hex [0ea3bb55c14aec8e074cd48c6a850526aa46965e176df53fadb433adeeeb56570000000000000000040000000000000000e1f5050000000001000000000000000000000000000000000000000000000000000000000000000305656d70747905656d70747900000100000000000000d101408a21fac7d536f667d84760bdd6bb8a2da0635422c0602d08c46b570a4d725863456cd1ede688fd77cea59c784c1d559687a1ada0df91e690e93c2adc8f7d42f218426974636f696e205369676e6564204d6573736167653a0a13526f6f6368205472616e73616374696f6e3a0a21028b2ed052fd389c09184a08fa8ffd1baa5ae5cdbf76f6369e4fd67e7b2af59adf406263727431707768756d7737326332686a6e7730633066367365643768783065306e687073366d75327767747638647761647165663777756a71376668767338]
```

Sign tx with reading file buffers:

```zsh
rooch transaction sign --read --file-location ~/text.txt --output
Sign transaction succeeded with the signed transaction hex [0ea3bb55c14aec8e074cd48c6a850526aa46965e176df53fadb433adeeeb56570000000000000000040000000000000000e1f5050000000001000000000000000000000000000000000000000000000000000000000000000305656d70747905656d70747900000100000000000000d101408a21fac7d536f667d84760bdd6bb8a2da0635422c0602d08c46b570a4d725863456cd1ede688fd77cea59c784c1d559687a1ada0df91e690e93c2adc8f7d42f218426974636f696e205369676e6564204d6573736167653a0a13526f6f6368205472616e73616374696f6e3a0a21028b2ed052fd389c09184a08fa8ffd1baa5ae5cdbf76f6369e4fd67e7b2af59adf406263727431707768756d7737326332686a6e7730633066367365643768783065306e687073366d75327767747638647761647165663777756a71376668767338]
```

- Sign tx and write to a file:

```zsh
rooch transaction sign --read --file-location ~/text.txt --file-destination ~/out.txt 
Write signed tx data succeeded in the destination
```

- Closes #2473

### CODE REVIEW

File: crates/rooch/src/commands/transaction/commands/sign.rs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                        ))
+                    })?;
+                signed_tx = context
+                    .sign(sender, tx_data.action, password, max_gas_amount)
+                    .await?;
+            } else {
+                return Err(RoochError::CommandArgumentError(
+                    "Argument --file-location is not provided".to_owned(),
+                ));
+            }
```
Comment (jolestar) [inline crates/rooch/src/commands/transaction/commands/sign.rs:77]: The `read` option is redundant. The file_location option is enough.

Comment (feliciss) [reply crates/rooch/src/commands/transaction/commands/sign.rs]: > The `read` option is redundant. The file_location option is enough.

Removed `read` option.

File: crates/rooch/src/commands/transaction/commands/sign.rs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                Ok(Some(signed_tx_hex))
+            } else {
+                println!(
+                    "Sign transaction succeeded with the signed transaction hex [{}]",
+                    signed_tx_hex
+                );
+
+                Ok(None)
+            }
+        } else if let Some(file_destination) = self.file_destination {
```
Comment (jolestar) [inline crates/rooch/src/commands/transaction/commands/sign.rs:98]: The `output` option is useless. Use `file_destination` is None

File: crates/rooch/src/commands/transaction/commands/sign.rs [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+    #[clap(flatten)]
+    tx_options: TransactionOptions,
+
+    #[clap(flatten)]
+    context: WalletContextOptions,
+
+    /// Read data from file, otherwise read transaction data hex from argument
+    #[clap(long, default_value = "false")]
+    read: bool,
```
Comment (jolestar) [inline crates/rooch/src/commands/transaction/commands/sign.rs:30]: The `read` can be removed.

File: crates/testsuite/features/cmd.feature [hunk 4]
```diff
@@ -67,6 +67,7 @@ Feature: Rooch CLI integration tests
       # transaction
       Then cmd: "transaction get-transactions-by-order --cursor 0 --limit 1 --descending-order false"
       Then cmd: "transaction get-transactions-by-hash --hashes {{$.transaction[-1].data[0].execution_info.tx_hash}}"
+      Then cmd: "transaction sign --tx-hex {{$.transaction[-1]}} --output --json
```
Comment (jolestar) [inline crates/testsuite/features/cmd.feature:70]: Need an assert to ensure the command execute success.

**Review by jolestar — APPROVED** [review]

**Comment by feliciss** [comment]
Next PR I'll squash this. Sorry for that.

**Pull request merged**